.circleci/real_config.yml

```diff
@@ -112,85 +112,74 @@ commands:
       - run: git submodule sync
       - run: git submodule update --init
 
-  skip-if-docs-only:
+  add-and-fetch-upstream:
     steps:
       - run: git remote add upstream https://github.com/determined-ai/determined
       - run: tools/scripts/retry.sh git fetch upstream
-      - run:
-          name: check for docs-only changes
-          command: |
-            MERGE_BASE="$(git merge-base upstream/main HEAD)"
-            DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
-            if [ "$DIFF_DIRS" = "docs" ] ; then
-              echo "docs-only change detected, halting job now"
-              circleci-agent step halt
-            else
-              echo "detected non-docs-only changes: $DIFF_DIRS"
-            fi
+
+  skip-if-dir:
+    parameters:
+      dir:
+        type: string
+      not:
+        type: boolean
+        default: false
+    steps:
+      - unless:
+          condition: << parameters.dir >>
+          steps:
+            - run:
+                name: skip if << parameters.dir >>-only changes
+                command: |
+                  MERGE_BASE="$(git merge-base upstream/main HEAD)"
+                  DIFF_DIRS="$(git diff-tree --no-commit-id --name-only "$MERGE_BASE" HEAD)"
+                  if [ "$DIFF_DIRS" = "<< parameters.dir >>" ] ; then
+                    echo "<< parameters.dir >>-only change detected, halting job now"
+                    circleci-agent step halt
+                  else
+                    echo "detected non-<< parameters.dir >>-only changes: $DIFF_DIRS"
+                  fi
+      - when:
+          condition: << parameters.dir >>
```

Comment:
Should this be whatever `parameters.not` is renamed to?

Reply:
ah yes. great catch!